### DISCUSSION TITLE
Strange `--section-div` behavior

### DISCUSSION THREAD

**Discussion by DavidBKentico** [jgm/pandoc#9559] [General]
We are using PandocRuby in a [Jekyll](https://jekyllrb.com/) project, and have encountered some confusing conversion behavior with headings, their ID attributes, and sections.

For a small minority of our headings, the surrounding `<div>` around a heading is replaced by a `<section>` tag, and the `id` attribute of the `<h>` tag is moved to the `<section>`. For example:

**Expected/correct example**:
```
<div class="dev-model-mvc" style="display: block;">
<h2 id="adding-css-styles-for-form-components">Adding CSS styles for form components
	<span title="Copy link to heading" class="heading-link-icon" onclick="copyHeadingLink(event)" onmouseout="copyIconMouseOut(event)"></span>
</h2>
<p>Use the following approach to add CSS styles for your form components:</p>
...
```
**Issue example**:
```
<section id="adding-scripts-and-styles-for-form-components" class="dev-model-core" style="display: block;">
<h2>Adding scripts and styles for form components
	<span title="Copy link to heading" class="heading-link-icon" onclick="copyHeadingLink(event)" onmouseout="copyIconMouseOut(event)"></span>
</h2>
<p>To add JavaScript and CSS styles required by your components, we recommend placing script and stylesheet files into sub-folders under:</p>
...
```

We haven't been able to 100% identify what causes this to happen, but all cases are headings that directly start after a wrapping `<div>` element. For example when using [Pandoc columns](https://pandoc.org/chunkedhtml-demo/10.6-columns.html) or our own custom tags that render divs.

This type of behavior seems to match the description of the [--section-divs](https://pandoc.org/MANUAL.html#options-affecting-specific-writers) Pandoc option. However, it's definitely not happening for all headings, or even all headings that are placed at the start of divs. Also, we definitely don't have this option enabled. When we tried enabling/adding `--section-divs`, then the described behavior happens for basically all headings.

Can you think of an explanation for this behavior, or a way to fix it?

Thanks

**Comment by jgm** [comment]
Please give an example of the input you're using, and the command (input format and options).

**Comment by DavidBKentico** [comment]
We run the following PandocRuby command:
```
PandocRuby.convert(content, {f: 'markdown-implicit_figures', to: :html}, '--wrap=none')
```
which should be equivalent to:
```
pandoc -f markdown-implicit_figures --to=html --wrap=none
```

We use markdown files as input. We also have custom ruby extensions (tags) in the source content, which get resolved into various types of HTML. So the input that gets passed to Pandoc looks something like this:
```
<div class="dev-model-core">

## Adding scripts and styles for form components

Text and other MD content + HTML

</div>
```

This is an example of a content extract where the heading issue occurs. If you want to see the full content, we prepared a "half-built" file with our custom tags processed into HTML, but before the Pandoc conversion. This is exactly what should be passed to Pandoc as input:
[developing-form-components.md](https://github.com/jgm/pandoc/files/14569550/developing-form-components.md)

**Comment by jgm** [comment]
See the documentation for the `native_divs` extension, and try `pandoc -t native` on this so you can see how pandoc is interpreting it. I think the solution for you would be to disable `native_divs`:  `-f markdown-native_divs`.  Or you could use a markdown dialect that doesn't have this, such as commonmark or gfm.

**Reply by DavidBKentico** [reply]
Thanks. Disabling the `native_divs` extension prevents the issue from happening in most cases. 

The only place (at least as far as we know so far) where it still occurs is if a heading is located immediately at the start of a [column](https://pandoc.org/chunkedhtml-demo/10.6-columns.html) container, For example something like:

```
::: columns ::::::

::: column-50

### Configure the environment for content editing
```

But this already points us in the right direction and helps a lot.
